alchemist-graphql/src/jvmMain/kotlin/it/unibo/alchemist/boundary/graphql/schema/model/surrogates/TimeSurrogate.kt

```diff
@@ -9,22 +9,62 @@
 
 package it.unibo.alchemist.boundary.graphql.schema.model.surrogates
 
+import com.expediagroup.graphql.generator.annotations.GraphQLDescription
+import com.expediagroup.graphql.generator.annotations.GraphQLIgnore
 import it.unibo.alchemist.model.Time
 
 /**
  * A GraphQL surrogate for a [Time].
  */
-enum class TimeSurrogate {
+data class TimeSurrogate(
+    @GraphQLIgnore override val origin: Time,
+) : GraphQLSurrogate<Time>(origin) {
+
+    /**
+     * The time represented as a double.
+     */
+    @GraphQLDescription("The time represented as a double")
+    val doubleTime: Double
+        get() = origin.toDouble()
+}
+
+/**
+ * A GraphQL input types used by clients for representing time.
+ */
+enum class TimeInput {
```

Comment:
What is this enum for?

Reply:
As long as I understood, we cannot create arbitrary time, but only through the companion object provided by `Time`. This enum is the 1-1 representation of that companion object, because GraphQL Kotlin does not consider companion objects when creating the schema.
In this way clients that has to specify a Time for some operations (e.g. `EnvironmentSurrogate.addNode`), would use this enum.

Reply:
@DanySK does it make sense?